docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -294,254 +296,285 @@ TODO
-  * So, at least in a first version, the lane info won't be shipped with TXs when broadcasting them
-  * TODO: we leave this bullet point here ATM, but we'll probably just deleted it
-* Before implementing lanes, is it better to first modularise the current mempool?
-  * we currently think improving modularisation is not gating for a mempool lanes MVP
-    * BUT, some things in the
-      _[modularity laundry list](https://www.notion.so/informalsystems/8a887f27e40b45dead689f2d1762b778?v=30a6fd81673b483fb16f4b7a9fc311fb)_
-      will be gating (e.g., `TxsAvailable`)
+This sections describes the architectural changes needed to implement an MVP of lanes in the
+mempool. The following is a summary of the key design decisions:
+- [[Lanes definition](#lanes-definition)] The list of lanes and their corresponding priorities
+  will be hardcoded in the application logic.
+- [[Internal data structures](#internal-data-structures)] There will be one concurrent list (CList) data structure per
+  lane.
+- [[Configuration](#configuration)] All lanes will share the same mempool configuration.
+- [[Adding transactions to the mempool](#adding-transactions-to-the-mempool)] When validating a transaction via CheckTx, the
+  application will optionally return a lane for the transaction.
+- [[Reaping transactions for creating blocks](#reaping-transactions-for-creating-blocks)] Transactions will be reaped from higher-priority
+  lanes first, preserving intra-lane FIFO ordering.
+- [[Transaction dissemination](#transaction-dissemination)] We will continue to use the current P2P channel for
+   disseminating transactions, and we will implement in the mempool the logic for selecting the
+   order in which to send transactions.
+
+### Lanes definition
+
+The list of lanes and their corresponding priorities will be hardcoded in the application logic. A
+priority is a value of type `uint32`, with 0 being a reserved value (see below). The application
+also needs to define which of the lanes it defines is the default lane. 
+
+To obtain the lane information from the application, we need to extend the ABCI `Info` response to
+include the following fields. These fields need to be filled by the application only in case it
+wants to implement lanes.
+```protobuf
+message InfoResponse {
+  ...
+  repeated uint32 lanes = 6;
+  uint32 default_lane = 7;
+}
+```
+Internally, the application may use `string`s to name lanes, and then map those names to priorities.
+The mempool does not care about the names, only about the priorities. That is why the lane
+information returned by the application only contains priorities.
+
+The lowest priority a lane may have is 1. The value 0 is reserved for two cases: when there are no
+priorities and for `CheckTx` responses of invalid transactions.
+
+On receiving the information from the app, CometBFT will validate that:
+- `lanes` has no duplicates (values in `lanes` don't need to be sorted),
```

Comment:
Would it make our life easier to require that the lanes are sorted (to simplify the validation code)?

Reply:
Not really. It's easy to sort the list and we wouldn't need to trust the app to do it.